google/resource_compute_global_forwarding_rule.go

```diff
@@ -141,6 +156,31 @@ func resourceComputeGlobalForwardingRuleCreate(d *schema.ResourceData, meta inte
 		return err
 	}
 
+	// If we have labels to update, try to set those too
+	if _, ok := d.GetOk("labels"); ok {
+		if computeApiVersion != v0beta {
```

Comment:
Could you extract part of this logic in its own method to avoid the duplication? The only difference I see is reading the fingerprint in create and the SetPartial.

Reply:
I tried, but even though there's duplication the function needs a lot of the surrounding state for context; the method signature looks like:

```go
func resourceComputeGlobalForwardingRuleSetLabels(config *Config, project, name string, labels map[string]string, labelFingerprint string) error {
```

With 5 parameters, I tend to think this logic is clearer duplicated twice... But if you disagree or know of a better way let me know.

Reply:
I would rather have this method than duplicate a block of 20 lines of code.

Reply:
I think it's more like 6-7 lines; I'll update the code and you can decide if it's improved or not.

Personally I tend to avoid factoring out functions that can't be named nicely or require a lot of surrounding context unless they're duplicated quite a lot.  Obviously there's a lot of subjectivity here so I'm fine either way.

Reply:
This is done, please take a look.